src/test/scala/is/hail/expr/ir/CompileSuite.scala

```diff
@@ -424,6 +426,8 @@ class CompileSuite {
     val aoff = f(region, start, false, stop, false, step, false)
     val actual = new UnsafeIndexedSeq(tRange, region, aoff)
     assert(actual.sameElements(expected))
+
+    TestUtils.interceptFatal("step size"){ f(region, 1, false, 5, false, 0, false) }
```

Comment:
CompileSuite is out, IRSuite (and friends) are in.  Move these testArrayRange tests to IRSuite.  @patrick-schultz has handy utility functions for checking exceptions.  If those aren't in yet, maybe he can PR them independently?

Reply:
I addressed all the other comments (including moving the tests above this test into InterpretSuite), but I've left this part in here for now because all the `eval` stuff does the `Interpret` and so would throw an error before checking the compiler. I can move when Patrick's stuff goes in?

Reply:
(ok, fixed with the new stuff).